sourcegraph/cody
# Autocomplete: add console-exporter for autocomplete OpenTelemetry tracing

## Context

- Replaces the deprecated `completionPostProcessLogger` with console-exporter for autocomplete OpenTelemetry tracing. It prints OpenTelemetry traces with their attributes and events to the console. It helps to debug the autocomplete data flow quickly.
- Next up: reuse the new logic for tracing-view-exporter to present the same information there.
- Builds on top of https://github.com/sourcegraph/cody/pull/2487 and https://github.com/sourcegraph/cody/pull/2486
- Closes https://github.com/sourcegraph/cody/issues/1921

## Test plan

1. Merge https://github.com/sourcegraph/cody/pull/2487 and https://github.com/sourcegraph/cody/pull/2486.
2. Set `cody.experimental.tracing` to `true`.
3. Connect standalone Chrome Dev tools.
4. Generate a code completion.
5. See traces printed to the console.
6. Disclaimer: `console.group()` titles are duplicated. It's a node.js bug: https://github.com/nodejs/node/issues/31973 
<img width="858" alt="Screenshot 2023-12-26 at 17 13 26" src="https://github.com/sourcegraph/cody/assets/3846380/28009e52-d1ff-4174-ad69-2fb445031d2a">
